numpy/core/src/umath/logical_gufuncs.c.src

```diff
@@ -256,79 +229,376 @@ static char types[] = {NPY_BYTE, NPY_BYTE, NPY_BOOL,
 
 
 /* -------------------------------------------------------------------------- */
-/* define name pointer to functions for each gufunc*/
-
-typedef struct gufunc_descriptor_struct {
-    char *name;
-    PyUFuncGenericFunction *funcs;
-} GUFUNC_DESCRIPTOR_t;
-
-static GUFUNC_DESCRIPTOR_t gufunc_descriptors [] = {
-    {"all_equal",           all_eq_funcs},
-    {"all_not_equal",       all_ne_funcs},
-    {"all_less",            all_lt_funcs},
-    {"all_less_equal",      all_le_funcs},
-    {"all_greater",         all_gt_funcs},
-    {"all_greater_equal",   all_ge_funcs},
-
-    {"any_equal",           any_eq_funcs},
-    {"any_not_equal",       any_ne_funcs},
-    {"any_less",            any_lt_funcs},
-    {"any_less_equal",      any_le_funcs},
-    {"any_greater",         any_gt_funcs},
-    {"any_greater_equal",   any_ge_funcs}
+/* create array of nulls for "data" for each gufunc type*/
+
+static void *array_of_nulls[] = {
+    (void *)NULL, (void *)NULL, (void *)NULL, (void *)NULL,
+    (void *)NULL, (void *)NULL, (void *)NULL, (void *)NULL,
+    (void *)NULL, (void *)NULL, (void *)NULL, (void *)NULL,
+    (void *)NULL, (void *)NULL, (void *)NULL, (void *)NULL
 };
 
 
 /* -------------------------------------------------------------------------- */
-/* create array of nulls for "data" for each gufunc type*/
-static void *array_of_nulls[] = {
-    (void *)NULL,
-    (void *)NULL,
-    (void *)NULL,
-    (void *)NULL,
-
-    (void *)NULL,
-    (void *)NULL,
-    (void *)NULL,
-    (void *)NULL,
-
-    (void *)NULL,
-    (void *)NULL,
-    (void *)NULL,
-    (void *)NULL,
-
-    (void *)NULL,
-    (void *)NULL,
-    (void *)NULL,
-    (void *)NULL
-};
+/* define docstrings*/
+
+static char const * const all_equal_doc =
+    "Return True if x1 == x2 for all elements along the last axis, False\n"
+    "otherwise.  Similar to (x1==x2).all(axis=-1), except the last dimension\n"
+    "of x1 and x2 must be equal and greater than 1.\n"
```

Comment:
> This code does already support broadcasting like that

Do you mean that you fiddle with the sizes before the ufunc, or just that it would support them if ufuncs did? I'm aware that this is a ufunc limititation, so is out of scope for this PR